robin/model/multimodal_encoder/timm_vision.py

```diff
@@ -10,65 +10,41 @@
 class TimmVisionTower(nn.Module):
     def __init__(self, vision_tower, args, delay_load=False):
         super().__init__()
-
+        assert args.mm_vision_select_layer == -1, "timm support output tokens of last layer only"
+        
+        self.vision_tower_name = vision_tower  # timm/vit_large_patch14_dinov2.lvd142m
+        self.select_layer = args.mm_vision_select_layer
+        self.select_feature = getattr(args, 'mm_vision_select_feature', 'patch')
+        self.hidden_size = None # Place Holder get this value in load_model function
+        self.dtype = None # Place Holder get this value in train function
         self.is_loaded = False
-
-        self.vision_tower_name = vision_tower
-        # self.select_layer = args.mm_vision_select_layer
-        # self.select_feature = getattr(args, 'mm_vision_select_feature', 'patch')
-
         if not delay_load:
             self.load_model()
-        else:
-            pass
-        
-        self.device = None
-        self.dtype = None
         
     def load_model(self):
-        
-        # print("VISION TOWER:", self.vision_tower_name)
-        if os.path.exists(self.vision_tower_name):
-            name = self.vision_tower_name.split("/")[-1]
-
-            # config = OPENCLIP_CONFIG_MAP[name] if name in OPENCLIP_CONFIG_MAP.keys() else name
-
-            # self.vision_tower, self.image_processor = create_model_from_pretrained(config, pretrained=self.vision_tower_name+'/open_clip_pytorch_model.bin')
-
-            self.vision_tower = timm.create_model(
-                name,
-                checkpoint_path=self.vision_tower_name+'/pytorch_model.bin',
-                num_classes=0,  # remove classifier nn.Linear
-            )
-            
-            data_config = timm.data.resolve_model_data_config(self.vision_tower)
-
-        else:
-            self.vision_tower = timm.create_model(
-                self.vision_tower_name,
-                pretrained=True,
-                num_classes=0,  # remove classifier nn.Linear
-            )
-            
-            data_config = timm.data.resolve_model_data_config(self.vision_tower)
-            
-        self.hidden_size = self.vision_tower.num_features#Can also use embed_dim
-        
-        #might need to change n shit. might not.
-        transforms = timm.data.create_transform(**data_config, is_training=True)
-        
-        self.image_processor = transforms
-        self.vision_tower.requires_grad_(False)
```

Comment:
If we remove this from the encoder side, then we need to check it in train instead.